ocw/lib/provider.py

```diff
@@ -31,48 +29,17 @@ def getData(self, name=None):
             return self.auth_json
         return self.auth_json[name]
 
-    def older_than_min_age(self, age):
-        return datetime.now(timezone.utc) > age + timedelta(
-            hours=PCWConfig.get_feature_property('cleanup', 'min-image-age-hours', self._namespace))
-
-    def needs_to_delete_image(self, order_number, image_date):
-        if self.older_than_min_age(image_date):
-            max_images_per_flavor = PCWConfig.get_feature_property('cleanup', 'max-images-per-flavor',
-                                                                   self._namespace)
-            max_image_age = image_date + timedelta(
-                hours=PCWConfig.get_feature_property('cleanup', 'max-image-age-hours', self._namespace))
-            return order_number >= max_images_per_flavor or max_image_age < datetime.now(timezone.utc)
-        else:
-            return False
-
-    def parse_image_name_helper(self, img_name, regex_s, group_key=['version', 'flavor', 'type', 'arch'],
-                                group_build=['kiwi', 'build']):
-        for regex in regex_s:
-            m = re.match(regex, img_name)
-            if m:
-                gdict = m.groupdict()
-                return {
-                    'key': '-'.join([gdict[k] for k in group_key if k in gdict and gdict[k] is not None]),
-                    'build': "-".join([gdict[k] for k in group_build if k in gdict and gdict[k] is not None]),
-                }
-        return None
-
-    def get_keeping_image_names(self, images):
-        images_by_flavor = dict()
-        for img in images:
-            if (img.flavor not in images_by_flavor):
-                images_by_flavor[img.flavor] = list()
-            images_by_flavor[img.flavor].append(img)
-
-        keep_images = list()
-        for img_list in [images_by_flavor[x] for x in sorted(images_by_flavor)]:
-            img_list.sort(key=lambda x: LooseVersion(x.build), reverse=True)
-            for i in range(0, len(img_list)):
-                img = img_list[i]
-                if (not self.needs_to_delete_image(i, img.date)):
-                    keep_images.append(img.name)
-
-        return keep_images
+    def older_than_max_age_hours(self, age):
+        """
+            older_than_max_age_hours - calculates if certain resource bypass maximum allowed TTL
+            maximum allowed TTL is controled by cleanup/max-age-hours pcw.ini config param
+            :param age: usually creation time of resource or any other timestamp which may be used to identify
+                age of the resource
+            :return: True if resource is already too old , false otherwise
+        """
+        delta_in_hours = PCWConfig.get_feature_property('cleanup', 'max-age-hours', self._namespace)
+        max_allowed_age = datetime.now(timezone.utc) - timedelta(hours=delta_in_hours)
+        return max_allowed_age > age
```

Comment:
The name of the method is `older_than_max_age_hours` but this comparison does exactly the opposite :thinking:

Reply:
There is a bit of confusion in this method - `age` is not an age (i.e. a duration) but a `timestamp`, i.e. a certain point in time. Suggestion:

```
def is_outdated(self, timestamp):
    delta = PCWConfig.get_feature_property('cleanup', 'max-age-hours', self._namespace)       # maximum allowed delta
    delta = timedelta(delta)
    now = datetime.now(timezone.utc)
    # Resource is outdated, if the given timestamp plus allowed delta exceeds the current timestamp
    return (timestamp + delta) > now
```

Reply:
the code which you suggesting will not give expected behavior . I can show you in example . 
Let's say `max-age-hours=7`  and if I call it with `is_outdated("01-01-2022 21-00")` it will gives me false . your method may be called `is_still_valid` because it will return true only for timestamps which didn't cross max-age-hours

Reply:
True, the last line should be `return now > (timestamp > delta)`